### DISCUSSION TITLE
Validating a whole object

### DISCUSSION THREAD

**Discussion by Funcan** [pydantic/pydantic#4956] [Question] [answered]
I want to validate that at least one of three fields is non-zero. I can do this by hooking up to the last field, but I don't see a way to access the earlier fields default values in the last validator, so code anything like the following fails. Obviously I can repeat myself and do `values.get("a", 0)` but then I need to remember to change the default in two places.

```
class foo(BaseModel):
  a = 0
  b = 1
  c = 2

  @validater("c", always=True):
  def non_zero(cls, v, values, *kwargs):
    if v == 0 and values["b"] == 0 and values["c"] == 0:
      raise ValueError("Can't have all zeros")

foo(b=0, c=0)
```

Any pointers greatly appreciate.

**Comment by duncan-thomas-hpe** [comment]
In case it is relevant: 
```
             pydantic version: 1.10.4
            pydantic compiled: True
                 install path: C:\Python311\Lib\site-packages\pydantic
               python version: 3.11.0 (main, Oct 24 2022, 18:26:48) [MSC v.1933 64 bit (AMD64)]
                     platform: Windows-10-10.0.19044-SP0
     optional deps. installed: ['typing-extensions']
```

**Comment by samuelcolvin** [comment, marked as answer]
You want [root validators](https://docs.pydantic.dev/usage/validators/#root-validators).

**Reply by Funcan** [reply]
Perfect, thanks. I was trying a root validator with `pre` which is of course not going to do what I want

**Reply by PeterSR** [reply]
As someone coming in from search engines in the future, how would this be done in Pydantic 2?
